src/controllers/activegate/reconciler/statefulset/reconciler.go

```diff
@@ -236,3 +261,22 @@ func (r *Reconciler) getDataFromCustomProperty(customProperties *dynatracev1beta
 	}
 	return customProperties.Value, nil
 }
+
+func (r *Reconciler) getDataFromAuthTokenSecret() (string, error) {
+	namespace := r.Instance.Namespace
+	var secret corev1.Secret
+	err := r.Get(context.TODO(), client.ObjectKey{Name: r.Instance.ActiveGateAuthTokenSecret(), Namespace: namespace}, &secret)
+	if err != nil {
+		return "", errors.WithStack(err)
+	}
+
+	dataBytes, ok := secret.Data[secrets.ActiveGateAuthTokenName]
+	if !ok {
+		return "", errors.Errorf("no auth-token found on secret '%s' on namespace '%s'", r.Instance.ActiveGateAuthTokenSecret(), namespace)
+	}
```

Comment:
refactor to use kubeobjects.GetSecret. Also, since this is a common use case, an addtional function in kubeobjects can be added to specifically extract the data and then be used here